client/app/guild/guild.js

```diff
@@ -76,14 +76,51 @@
 
         socket.forward('get:guild',$scope);
         $scope.$on('socket:get:guild',function(ev,guild){
-            $scope.$parent.loading = false;
             //If not exit, redirect user to dashboard
             if(guild===null)
                 $state.go("dashboard");
             $scope.guild = guild;
+
+            socket.emit('get:userGuildRank',{"region":$stateParams.region,"realm":$stateParams.realm,"name":$stateParams.name});
+
+            socket.forward('get:userGuildRank',$scope);
+            $scope.$on('socket:get:userGuildRank',function(ev,rank){
+                $scope.$parent.loading = false;
+                $scope.guildRank = rank;
+
+                if (rank === 0) {
+                    // This is the guild leader, put the rank permissions in an easier form for table rendering
+                    var perms = $scope.guildRankPerms = [];
+                    var isOfRank = function(i) {
+                        return function (member) { return member.rank === i; };
+                    };
+                    for (var i = 0; i < 10; i++) {
```

Comment:
Hard-coded `10` here for number of ranks including the GM. Not sure if we want to parameterize this or where it would go, it's unlikely they'll change the number of guild ranks.